tests/python/test_bit_struct.py

```diff
@@ -77,6 +77,34 @@ def verify_val(idx: ti.i32):
         verify_val.__wrapped__(idx)
 
 
+@ti.test(require=ti.extension.quant, debug=True, cfg_optimization=False)
+def test_custom_int_full_struct():
+    cit = ti.type_factory_.get_custom_int_type(32, True)
+    x = ti.field(dtype=cit)
+    ti.root.dense(ti.i, 1)._bit_struct(num_bits=32).place(x)
+
+    @ti.kernel
+    def set_val():
+        x[0] = 15
+
+    @ti.kernel
+    def varify_val1():
+        assert x[0] == 15
+
+    @ti.kernel
+    def set_val2():
+        x[0] = 12
+
+    @ti.kernel
+    def varify_val2():
+        assert x[0] == 12
+
+    set_val()
+    varify_val1()
+    set_val2()
+    varify_val2()
```

Comment:
This case will fail in current implementation. Because 32 times of left shifting of a 32-bits `1` will be carried on.